tests/test_service.py

```diff
@@ -0,0 +1,256 @@
+import sys
+from configparser import ConfigParser
+from dxlbootstrap.util import MessageUtils
+from dxlclient import Request
+from dicttoxml import dicttoxml #pylint: disable=import-error
+from dxlurlvoidservice import UrlVoidApiService
+from dxlurlvoidservice.requesthandlers import UrlVoidApiCallback
+
+from tests.test_base import BaseClientTest
+from tests.test_value_constants import *
+from tests.mock_urlvoidhttpserver import MockServerRunner
+
+sys.path.append(
+    os.path.dirname(os.path.abspath(__file__)) + "/../.."
+)
+
+def create_service_configfile(config_file_name):
+    config = ConfigParser()
+
+    config['General'] = {'apiKey': SAMPLE_API_KEY}
+
+    with open(config_file_name, 'w') as config_file:
+        config.write(config_file)
+
+
+class TestConfiguration(BaseClientTest):
+
+    def test_loadconfig(self):
+
+        create_service_configfile(
+            config_file_name=URLVOID_CONFIG_FILENAME,
+        )
+
+        urlvoid_service = UrlVoidApiService(TEST_FOLDER)
+        urlvoid_service._load_configuration()
+
+        self.assertEqual(urlvoid_service.api_key, SAMPLE_API_KEY)
+
+        os.remove(URLVOID_CONFIG_FILENAME)
+
+
+    def test_registerservices(self):
+        with MockServerRunner():
+
+            create_service_configfile(
+                config_file_name=URLVOID_CONFIG_FILENAME
+            )
+
+            with BaseClientTest.create_client(max_retries=0) as dxl_client:
+                dxl_client.connect()
+
+                vt_service = UrlVoidApiService(TEST_FOLDER)
+                vt_service._dxl_client = dxl_client
+
+                vt_service._load_configuration()
+                vt_service.on_register_services()
+
+                self.assertTrue(len(vt_service._services) > 0)
+
+                expected_vt_topics = {
+                    UrlVoidApiService.REQ_TOPIC_HOST_INFO,
+                    UrlVoidApiService.REQ_TOPIC_HOST_RESCAN,
+                    UrlVoidApiService.REQ_TOPIC_HOST_SCAN,
+                    UrlVoidApiService.REQ_TOPIC_STATS_REMAINED,
+                }
+
+                for expected_topic in expected_vt_topics:
```

Comment:
Could we check here that the `expected_vt_topics` set equals `vt_service._services[0].topics` set rather than checking that `vt_service._services[0].topics` is a superset of `expected_vt_topics`?